libs/category/state/src/effects/category-page-filter.effects.ts

```diff
@@ -60,53 +57,47 @@ export class DaffCategoryPageFilterEffects<
     private actions$: Actions,
     @Inject(DaffCategoryDriver) private driver: DaffCategoryServiceInterface<V, W>,
 		@Inject(DAFF_CATEGORY_ERROR_MATCHER) private errorMatcher: ErrorTransformer,
-    private store: Store<any>,
+    private facade: DaffCategoryFacade,
   ){}
 
-	private categorySelectors = getDaffCategorySelectors<V, W>();
-
+  /**
+   * Updates the filters applied to the category page. It will take the currently
+   * applied filters from state, form them into a request, and attempt to apply
+   * those request to a backend service.
+   *
+   * @param debounceFrame the amount of time to delay when apply/removing filters
```

Comment:
basically for rate throttling. If you quickly dispatch a bunch of requests then it will only actually send requests to the driver once every interval defined by the `debounceFrame`. Note that `throttle` is used here instead of `debounce` because `throttle` sends an initial request (when that option is configured) whereas `debounce` does not allow for this behavior.

Reply:
see the test case for more info

Reply:
78713eeb

Reply:
can't this be handled with mergeMap instead?

Reply:
I don't think so. `mergeMap` sends every request. I think we specifically don't want to send every request since every category request has all the info needed to fully describe the category.

Reply:
That makes a lot of sense. Last thing is that I was looking at the debounceFrame docs and I think we don't want the `trailing` config to be true. I think that will make it send the first two requests (within 100ms) in quick succession, then wait 100ms to send the third. It looked like this config is mostly for differentiating between single and double clicks, so that if you didn't want a single click to go through, you'd set `{ leading: false, trailing: true }`. I'm not sure what the usecase for `{leading: true, trailing: true }` would be. So I think we just want the default config values for this.

Reply:
In retrospect, after considering this, I think we actually want `debounceTime` as I originally had thought, but with a fairly low debounce time (150ms) the goal here is to lower the total requests sent as users apply, update and remove filters. If a user clicks:

"Apply, Remove, Apply" in quick succession this is the same as a single "apply" so it really only makes sense to wait until the user's interactions have completed to send the request (e.g. a debounce) as opposed to a throttle.

Reply:
Consider: 

1. https://stackblitz.com/edit/wyjuke--run?file=index.ts
2. https://stackblitz.com/edit/7cjm9c--run?file=index.ts

Reply:
> I think we actually want debounceTime as I originally had thought

I disagree, why not send the request immediately?

> it really only makes sense to wait until the user's interactions have completed to send the request

Sounds like an app sided concern. I think our lib should aim to be as performant as possible. We have spent effort optimizing functions with way way less than 150ms to gain.

Reply:
Although looking at your examples, it seems `throttleTime` also doesn't have ideal behavior. The most ideal behavior to me seems to be if `debounceTime` would send the first request.

Reply:
I still think `throttleTime` makes more sense, because we shouldn't slow down every request by 150ms just because there might be an extremely rare case where the user is clicking category requests really quickly for an extended period of time. Even then, the exposure to the backend is throttled. 

My only concern is with this part of the `throttleTime` docs: 
NOTE: I just tested this and it turned out the docs are just wrong. Read on if you're interested. Here is the stackblitz showing the docs aren't correct: https://stackblitz.com/edit/w5iibw-kswnee?file=index.ts
```
const throttleConfig = {
   leading: true,
   trailing: true
 }
In this example, the output would be the primary click and
the double click, but restricts additional clicks within 400ms.
```
I sort of paraphrased that, but that's the spirit of what the docs are saying. This is why I originally thought that using `trailing: true` wasn't correct for our usecase. It sounds like the second click will be let through immediately, ignoring the `debounceFrame`, but that isn't the case. It waits till the timer completes, then releases the second click. But if that's true, then do they mean that any clicks after the second one get canceled completely? So we'd need to test that to make sure the docs are actually wrong.

Also, @griest024, it might seem like `throttleTime` doesn't work as you'd expect because stackblitz is funny about how it saves files and updates the app that's running. I had to close the app and reopen it to get it to actually update the code each time. Unless I'm missing something, I think `throttleTime` does have the behavior we want. Here is a different stackblitz that I think showcases that: https://stackblitz.com/edit/w5iibw-kswnee?file=index.ts

Reply:
> there might be an extremely rare case where the user is clicking category requests really quickly

Yeah I think 99% of the time the actions won't be coming in fast enough for the 150ms time to ever be a factor.

> It waits till the timer completes, then releases the second click. But if that's true, then do they mean that any clicks after the second one get canceled completely? So we'd need to test that to make sure the docs are actually wrong.

See the unit test. If 3 events are dispatched within the interval, `throttleTime` sends the first one immediately, and sends the third one at the end of the interval. The second one is discarded. Since all category requests contain all info this shouldn't be an issue.

> it might seem like throttleTime doesn't work as you'd expect because stackblitz is funny about how it saves files and updates the app that's running. 

That could be the case. From the unit testing that @damienwebdev and I did, it really seemed like `throttleTime` had the correct behavior.

Reply:
> See the unit test. If 3 events are dispatched within the interval, throttleTime sends the first one immediately, and sends the third one at the end of the interval. The second one is discarded. Since all category requests contain all info this shouldn't be an issue.

I don't think the unit tests actually test what I was talking about. I might be misreading the test, because honestly I find the testscheduler tests confusing as hell (in general). But you'd need to have the second request put the filters in a different state than the third request. If the second and third requests are the same, then there's no way to tell if the trailing call was with the second or third request. But I don't think the test necessarily needs written, because I tested it with the stackblitz.

Reply:
oh you're right, I see. yeah your stackblitz verifies that behavior, I don't think we need it in our unit tests

Reply:
I think we're all good on this now. After speaking with Damien, the plan moving forward is to go with `throttleTime` for now, then possibly write another operator in the future that is a debounceTime/throttleTime hybrid.